examples/with-remix/app/root.tsx

```diff
@@ -53,12 +60,41 @@ export default function App() {
   const { ENV } = useLoaderData<LoaderData>();
 
   // Remix modules cannot have side effects so the initialization of `wagmi`
-  // client is abstracted away in a `useEffect`.
+  // client happens during render, but the result is cached via `useState`
+  // and a lazy initialization function.
   // See: https://remix.run/docs/en/v1/guides/constraints#no-module-side-effects
-  const { client, chains } = useSetupWagmi({
-    appName: 'RainbowKit Remix Example',
-    alchemyId: ENV.ALCHEMY_ID,
-    enablePublicTestnets: ENV.PUBLIC_ENABLE_TESTNETS,
+  const [{ client, chains }] = useState(() => {
```

Comment:
Is there a reason to use `useState` over `useMemo`?

Reply:
The initializer for `useState` is guaranteed to only run once, while `useMemo` doesn't technically guarantee this.

From https://reactjs.org/docs/hooks-reference.html#usememo:
> **You may rely on useMemo as a performance optimization, not as a semantic guarantee.** In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without `useMemo` — and then add it to optimize performance.

Reply:
True!